2017/proposals/ranasinghe-kanchana.md

```diff
@@ -0,0 +1,156 @@
+
+### June 5th - June 9th
+* color-bar support for imshow
+
+The color bar is mapped from numerical data for heat-maps currently, creating boundaries for each class. Due to discrete nature of categorical data, ListedColormaps will be used and the boundaries required for each distinct stage will be extended to support non-numerical values. The string values of categorical data will be directly mapped to each distinct color level by the units integration. 
+
+### June 12th - June 16th
+* legend support for imshow
+
+This is an additional improvement attempted. Since heat-maps do not currently have a direct method for implementation of a legend, this will focus on extending the legend support of other scalarMappables. Also the categorical support for legend here will mostly be for the purpose of serving as a foundation for legend support for other scalarMappables. 
+
+### June 19th - June 23th, **End of Phase 1**
+* testing and debugging of imshow function (heat-map API)
+
+### June 26 - June 30th, **Begin of Phase 2**
+* framework outline for norm functionality extensions
+* norm basic functionality including tests and documentation
+
+With regards to the normalization function, in a Matplotlib sense, it encompasses a function mapping different data values to different colors (or color levels). Currently, the different data values to be mapped from are expected to be numerical. This limitation will be worked around to enable support for categorical data (string data). Also during the course of the project focus will be given to the currently available discrete normalization methods (matplotlib.colors.BoundaryNorm) which handle data in a different sense which may be used as a foundation for this additional unit support. 
+
+### July 3rd - July 7th
+* framework outline for cmap functionality extensions
+* cmap basic functionality including tests and documentation
+
+Currently, the process of using matplotlib.pyplot for heat-maps and other scalarMappables includes the creation of a color-map followed by normalization of all available data to fit this color-map. Color-maps plotting functionality is threefold: sequential, divergent and qualitative. Considering the intrinsic discreteness of categorical data, the extension of the color-maps API will focus on its qualitative plotting ability. However the fundamental behavior of the color-maps API will not require any changes for handling categorical data as our approach simply involves the integration of unit support. Therefore it is possible to build this extended support for categorical data while keeping the public API intact. 
+
+### July 10th - July 14th
+* integration of norm and cmap into heatmaps
```

Comment:
So if we call  a function like imshow giving only data as input, allowing the default norm and cmap to be used, when the data is categorical it will not automatically identify this and use norm appropriate at the end of my work in stage one. (Correct me if I'm wrong) User will be required to call relevant color function to create appropriate norm and cmap and then use this on imshow. This is because at that stage cmap and norm APIs are not extended to support categorical data. What is done here is making required changes to ensure imshow uses all new functionality added to norm and cmap. 

But again since I will move the task of creating norm and cmap to the top (phase 1) and push imshow API categorical support to phase 2, this will most probably be not required as it can be integrated to imshow from the beginning (if required).